src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala

```diff
@@ -25,6 +90,62 @@ object Directives extends AuthDirectives with LeaderDirectives with AkkaDirectiv
     }
   }
 
+  private val marathonApiKeywords = Set("restart", "tasks", "versions")
+
+  //basically it's equivalent to path.takeWhile(s => !marathonApiKeywords.contains(s)) map createPathId
```

Comment:
Where is the difference?

Reply:
Unfortunately `path` is not even a `Traversable`, so recursion is required.